components/header/wallet_modal.tsx

```diff
@@ -0,0 +1,125 @@
+import { Dispatch, FormEvent, SetStateAction, useState } from 'react';
+import Image from 'next/image';
+import ReactModal from 'react-modal';
+
+import { validateWalletAddress } from '@utils';
+
+import arconnectLogo from '../../public/arconnect.png';
+import {
+	CloseModalButton,
+	ImportWalletErrorMessage,
+	ImportWalletForm,
+	ImportWalletInput,
+	ModalButton,
+	ModalButtons,
+	ModalContent,
+	ModalText,
+} from './wallet_modal.style';
+
+const WalletModal = ({
+	isOpen,
+	closeModal,
+	setWalletAddress,
+}: {
+	isOpen: boolean;
+	closeModal: Function;
```

Comment:
I think `Function` is a very wide type here, allowing for any parameters or return types. It doesn't give the consumer of this component much type information. Consider `() => void` here

Reply:
https://github.com/ardriveapp/inferno-leaderboard/pull/3/commits/226ed0da21bfae76302703985cbbef8cff1deb47